mmdet/models/anchor_heads/anchor_head.py

```diff
@@ -45,6 +47,7 @@ def __init__(self,
                  loss_bbox=dict(
                      type='SmoothL1Loss', beta=1.0 / 9.0, loss_weight=1.0)):
         super(AnchorHead, self).__init__()
+        # FIXME: in_channels is never used!
```

Comment:
I'm unclear on the difference between `in_channels` and `feat_channels`. Can you explain more?

Reply:
- `in_channels` is the input feature map channel, e.g., the feature map from FPN or backbone.
- `feat_channels` is the output channel after some convs, e.g., the 3x3 rpn_cpnv, or 4 consecutive convs in RetinaHead.

Reply:
So that makes sense for RetinaHead, `feat_channels` is some number of hidden channels in the intermediate convolutions and `in_channels` are the number of channels being passed to `RetinaHead.forward`. 

However, in AnchorHead, `feat_channels` is used as the input channels for the first convolution and `in_channels` is unused. In this there are no hidden channels and if the number of input channels is different than `feat_channels` an error will be raised. 

Based on your definitions, I think the `in_channels` and `feat_channels` argument in the `AnchorHead` module should be swapped. Its also just weird that `in_channels` is a required argument, but its unused.

This looks like it is also a problem in `GuidedAnchorHead`, but all other anchor head clases seem to use `in_channels` in the expected way. Looking at the configs it looks like `AnchorHead` and `GuidedAnchorHead` aren't explicitly used, so I think this bug is just an oversight. Patching the implementation to use `in_channels` instead of `feat_channels` as the first convolution input channels would fix the issue and I don't think it would break anything. Would you like me to do that in this PR or make it a separate issue (or perhaps I'm being very dense and everything is correct but I still don't understand the issue)?

Reply:
`AnchorHead` and `GuidedAnchorHead` are just base classes, and all inheriting classes are expected to override `_init_layers()`, which will use one or more convs to transform the features from `in_channels` to `feat_channels`.

It's indeed misleading here, and it should be ok to use `feat_channels` with `in_channels`. A separate PR may be raised for fixing that.